codeSnippets/snippets/tutorial-server-docker-compose/Dockerfile

```diff
@@ -1,5 +1,25 @@
-FROM openjdk:11.0.16
+# Stage 1: Cache Gradle dependencies
+FROM gradle:latest AS cache
+RUN mkdir -p /home/gradle/cache_home
+ENV GRADLE_USER_HOME /home/gradle/cache_home
+COPY build.gradle.* gradle.properties /home/gradle/java-code/
+WORKDIR /home/gradle/java-code
+RUN gradle clean build -i --stacktrace
+
+# Stage 2: Build Application
+FROM gradle:latest AS build
+COPY --from=cache /home/gradle/cache_home /home/gradle/.gradle
+COPY . /usr/src/java-code/
+WORKDIR /usr/src/java-code
+COPY --chown=gradle:gradle . /home/gradle/src
+WORKDIR /home/gradle/src
+# Build the fat JAR, Gradle also supports shadow
+# and boot JAR by default.
+RUN gradle buildFatJar --no-daemon
+
+# Stage 3: Create the Runtime Image
+FROM openjdk:22 AS runtime
```

Comment:
Why do we still use the `openjdk` image in our examples if it's deprecated?

Reply:
I dont exactly know where the users should be referred to. OpenJDK is deprecated yes, but if I write something like amazoncorretto there, its still open to discussion like “amazoncorretto might be bad X is better”. I thought keeping a general solution and leaving a deprecated warning is better

Reply:
I'd choose one alternative, refer to the other possibilities and remove the depreciation warning.

Reply:
Alrighty almighty. I will go with Amazon Corretto, it works without issues on all architectures, arm64 etc

Reply:
I suggest `This example uses the Amazon Coretto Docker Image, but you can substitute it with any other suitable alternative, such as the following:`

Reply:
and remove Amazon Coretto from the list